### PR TITLE
[Kotlin Migration] StudyOptionsFragment

### PR DESCRIPTION
**Pull request by Akshay0701**

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/StudyOptionsFragment.kt [hunk 1]
```diff
@@ -0,0 +1,719 @@
...
+        parent!!.removeAllViews()
+        parent.addView(newView)
+    }
+
+    private val mUndoListener: TaskListener<Unit, Computation<NextCard<*>?>?> = object : TaskListener<Unit, Computation<NextCard<*>?>?>() {
+        override fun onPreExecute() {}
+        override fun onPostExecute(v: Computation<NextCard<*>?>) {
+            openReviewer()
+        }
+    }
```
Comment (Akshay0701) [inline AnkiDroid/src/main/java/com/ichi2/anki/StudyOptionsFragment.kt:251]: all other conversion is right, but the problem occurring here, what should I do?

Comment (david-allison-1) [reply AnkiDroid/src/main/java/com/ichi2/anki/StudyOptionsFragment.kt:251]:
```
    private val mUndoListener: TaskListener<Unit, ComputeResult> = object : TaskListener<Unit, ComputeResult>() {
        override fun onPreExecute() {}
        override fun onPostExecute(result: ComputeResult) {
```

You also likely need a `@JvmOverloads` for `refreshInterface`

Comment (Akshay0701) [reply AnkiDroid/src/main/java/com/ichi2/anki/StudyOptionsFragment.kt:251]: Thanks! cleared all errors

**Review by david-allison-1 — CHANGES_REQUESTED** [review]
```
    var mOnRequestReviewActivityResult = registerForActivityResult(ActivityResultContracts.StartActivityForResult()) { result: ActivityResult ->
Fix the issues identified by lint, or add the following to your build script to proceed with errors:
    ^
...
/home/runner/work/Anki-Android/Anki-Android/AnkiDroid/src/main/java/com/ichi2/anki/StudyOptionsFragment.kt:475: Error: Variable name should not use field prefixes. [VariableNamingDetector from com.ichi2.anki.lint]
android {
        /**
    lintOptions {
        ^
/home/runner/work/Anki-Android/Anki-Android/AnkiDroid/src/main/java/com/ichi2/anki/StudyOptionsFragment.kt:479: Error: Variable name should not use field prefixes. [VariableNamingDetector from com.ichi2.anki.lint]
        abortOnError false
        /**
    }
}
...
        ^
/home/runner/work/Anki-Android/Anki-Android/AnkiDroid/src/main/java/com/ichi2/anki/StudyOptionsFragment.kt:483: Error: Variable name should not use field prefixes. [VariableNamingDetector from com.ichi2.anki.lint]
        /**
        ^
/home/runner/work/Anki-Android/Anki-Android/AnkiDroid/src/main/java/com/ichi2/anki/StudyOptionsFragment.kt:487: Error: Variable name should not use field prefixes. [VariableNamingDetector from com.ichi2.anki.lint]
        /**
        ^
/home/runner/work/Anki-Android/Anki-Android/AnkiDroid/src/main/java/com/ichi2/anki/StudyOptionsFragment.kt:491: Error: Variable name should not use field prefixes. [VariableNamingDetector from com.ichi2.anki.lint]
        /**
        ^
/home/runner/work/Anki-Android/Anki-Android/AnkiDroid/src/main/java/com/ichi2/anki/StudyOptionsFragment.kt:495: Error: Variable name should not use field prefixes. [VariableNamingDetector from com.ichi2.anki.lint]
        /**
        ^
7 errors, 0 warnings
```

File: AnkiDroid/src/main/java/com/ichi2/anki/StudyOptionsFragment.kt [hunk 2]
```diff
@@ -1,3 +1,4 @@
+//noinspection MissingCopyrightHeader #10088
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/StudyOptionsFragment.kt:1]: Why? There is a copyright header

File: AnkiDroid/src/main/java/com/ichi2/anki/StudyOptionsFragment.kt [hunk 3]
```diff
@@ -11,754 +12,712 @@
...
      * Show the context menu for the custom study options
      */
-    private void showCustomStudyContextMenu() {
-        final AnkiActivity ankiActivity = ((AnkiActivity) requireActivity());
-        CustomStudyDialog d = FragmentFactoryUtils.instantiate(ankiActivity, CustomStudyDialog.class);
-        d.withArguments(CustomStudyDialog.CONTEXT_MENU_STANDARD, getCol().getDecks().selected());
-        ankiActivity.showDialogFragment(d);
+    private fun showCustomStudyContextMenu() {
+        val ankiActivity = requireActivity() as AnkiActivity
+        val d = instantiate(ankiActivity, CustomStudyDialog::class.java)
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/StudyOptionsFragment.kt:235]: cleanup: give it a better name

File: AnkiDroid/src/main/java/com/ichi2/anki/StudyOptionsFragment.kt [hunk 4]
```diff
@@ -11,754 +12,712 @@
...
 
-    private void refreshInterfaceAndDecklist(boolean resetSched) {
-        refreshInterface(resetSched, true);
+    fun refreshInterface() {
+        refreshInterface(resetSched = false, resetDecklist = false)
     }
 
-    protected void refreshInterface() {
-        refreshInterface(false, false);
+    private fun refreshInterfaceAndDecklist(resetSched: Boolean) {
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/StudyOptionsFragment.kt:451]: KotlinCleanup: `default value + JvmOverloads`

File: AnkiDroid/src/main/java/com/ichi2/anki/StudyOptionsFragment.kt [hunk 5]
```diff
@@ -11,754 +12,712 @@
...
+        mLoadWithDeckOptions = false
+        return true
     }
 
-    private @Nullable Collection getCol() {
-        try {
-            return CollectionHelper.getInstance().getCol(getContext());
-        } catch (Exception e) {
-            // This may happen if the backend is locked or similar.
+    // This may happen if the backend is locked or similar.
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/StudyOptionsFragment.kt:656]: comment was duplicated

Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/StudyOptionsFragment.kt:656]: comment was duplicated

**Review by david-allison-1 — APPROVED** [review]
LGTM, thansk

**Review by mikehardy — APPROVED** [review]

**Pull request merged**